### PR TITLE
Updated search

### PR DESCRIPTION
**Pull request by srietkerk**

### CODE REVIEW

File: frontend/src/HikeFinder.js [hunk 1]
```diff
@@ -135,30 +127,39 @@ function HikeFinder(props) {
...
-    if (indices.size === 0) {
+    console.log(hikeData)
+
+    if (input.length === 0 && finderState.recentlySearched) {
+      indices = new Set([...finderState.beforeSearchIndices])
+      searched = false;
+    }
+
+    else if (indices.size === 0) {
       console.log("Nothing matched your search results, please try something else!")
```
Comment (Baagi) [inline frontend/src/HikeFinder.js:150]: > console.log("Nothing matched your search results, please try something else!")

This is only logged in the browser?

File: frontend/src/components/FilterBar.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { PropTypes } from 'prop-types';
 function FilterBar(props) {
   return (
     <div id="filterBar">
-      <SearchBar keyword={props.keyword} setKeyword={props.setKeyword} />
+      <SearchBar updateBySearch={props.updateBySearch}/>
```
Comment (Baagi) [inline frontend/src/components/FilterBar.js:11]: Nice! Looks more understandable now.

**Review by MaggieYang57 — COMMENTED** [review]
Hi Sarah! The code seems pretty concise and understandable. The variables and function names are also very clear and consistent with our style guide, which is great. I personally don't know the details of how the search bar is implemented, so maybe it would be useful to add comments in SearchBar.js to describe the functions a little more for other developers. Other than that, the code looks good to me!

File: frontend/src/components/SearchBar.js [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+    timeout = setTimeout(later, wait);
+  };
+};
+
 function SearchBar(props) {
+  const [inputVal, setInputVal] = useState('');
+  const [debounceFunc, setFunc] = useState(null);
+
+  useEffect(() => {
+    const debFn = debounce(props.updateBySearch, 500);
```
Comment (Baagi) [inline frontend/src/components/SearchBar.js:24]: I'm not familiar with debounce. What does the 500 refer to?

**Comment by Baagi** [comment]
Looks good to me! Next time please commit eslint in a separate commit so we can find the new code faster.

File: frontend/src/HikeFinder.js (hunk 1 above)
Comment (srietkerk) [reply frontend/src/HikeFinder.js:150]: Hey Bat! I definitely will change this. This is my first solution to responding to the user if a search result doesn't find anything. In the future, I am hoping to have a response on the frontend like a pop-up or just a text response under the search bar.

File: frontend/src/components/FilterBar.js (hunk 2 above)
Comment (srietkerk) [reply frontend/src/components/FilterBar.js:11]: Thanks!

File: frontend/src/components/SearchBar.js (hunk 3 above)
Comment (srietkerk) [reply frontend/src/components/SearchBar.js:24]: That's a good point, I might want to implement a sort of global variable to get rid of the hard-coded number there. The 500 stands for the number of milliseconds the search bar (based on the debouncer) is going to wait for the user to stop typing. After that amount of time, the debouncer is going to call the function updateBySearch.

**Comment by srietkerk** [comment]
> Hi Sarah! The code seems pretty concise and understandable. The variables and function names are also very clear and consistent with our style guide, which is great. I personally don't know the details of how the search bar is implemented, so maybe it would be useful to add comments in SearchBar.js to describe the functions a little more for other developers. Other than that, the code looks good to me!

That sounds like a great idea. Since the code is already merged,  I will definitely add comments when I push code next.

**Pull request merged**